src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/QueueHealthDecider.java

```diff
@@ -91,11 +92,11 @@ private void configureActionPriority() {
    * <p>Action relevance decided based on user configured priorities for now, this can be modified
    * to consume better signals going forward.
    */
-  private Action computeBestAction(NodeKey esNode, ResourceEnum threadPool) {
+  private Action computeBestAction(NodeKey esNode, Resource resource) {
     Action action = null;
     for (String actionName : actionsByUserPriority) {
       action =
-          getAction(actionName, esNode, threadPool, getNodeQueueCapacity(esNode, threadPool), true);
+        getAction(actionName, esNode, resource.getResourceEnum(), getNodeQueueCapacity(esNode, resource), true);
```

Comment:
resource object read from QueueRejctionRca is threadpool + queue rejection. That is not the metric we want to read here. What we want to read from node config cache is threadpool + queue capacity